r/R/trajectory_analysis.R

```diff
@@ -0,0 +1,127 @@
+#' Generate UMAP and nodes coordinates for the initial trajectory analysis plot
+#'
+#' Returns a json object with UMAP and nodes coordinates, to be used for the initial plot (embedding + trajectory nodes)
+#'
+#' This represents the first step of the trajectory analysis. It allows the creation of the initial trajectory analysis plot, which is composed of an embedding and starting nodes (vertices in the lines).
+#' This plot is used to select the root nodes among the starting nodes. The root nodes will be then used in the following step of the trajectory analysis for pseudotime calculation.
+#'
+#' @param data SeuratObject
+#'
+#' @return a json with nodes and umap coordinates
+#' @export
+runGenerateTrajectoryGraph <- function(data) {
```

Comment:
I think this should be `function(req, data)` because the parameters passed to it (see https://github.com/biomage-ltd/worker/blob/0de12af5a7abf8735c6944b38ae427d29df2eb6d/r/work.R#L57) is `(req, data)`

Reply:
that's a good catch!